lib/interceptor.js

```diff
@@ -158,12 +158,11 @@ Interceptor.prototype.reqheaderMatches = function reqheaderMatches(options, key)
     //  NOTE: We use lower-case header field names throughout Nock.
     if (key === 'host' &&
         (_.isUndefined(header) ||
-        _.isUndefined(reqHeader)))
-    {
+            _.isUndefined(reqHeader))) {
         return true;
     }
 
-    if (!_.isUndefined(reqHeader) && !_.isUndefined(header)) {
+    if (reqHeader && header) {
```

Comment:
ooops that one should not have been changed :) Sorry my fault that it slipped, I’ll fix it in a follow up PR